### PR TITLE
Adds tab pane to the data/style form

### PR DESCRIPTION
**Pull request by javierarce**
First task of #6511

tabpane

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js [hunk 1]
```diff
@@ -32,6 +32,6 @@ module.exports = cdb.core.View.extend({
   },
 
   _onButtonClicked: function () {
-    this.model('selected', true);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js]: oh, how did we miss that? :|

Comment (viddo) [inline lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js]: missing test… :joy:

**Comment by javierarce** [comment]
CR: @xavijam 
/cc @viddo

File: lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js]: Yep, totally :+1:

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-data-view.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  initialize: function (opts) {
+    if (!opts.tableModel) throw new Error('tableModel is required');
+    if (!opts.widgetDefinitionModel) throw new Error('widgetDefinitionModel is required');
+    this._tableModel = opts.tableModel;
+    this._widgetDefinitionModel = opts.widgetDefinitionModel;
+  },
+
+  render: function () {
+    this.clearSubViews();
+    this.$el.html('');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-data-view.js]: `this.$el.empty();`

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-style-view.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+module.exports = cdb.core.View.extend({
+  initialize: function (opts) {
+    if (!opts.widgetDefinitionModel) throw new Error('widgetDefinitionModel is required');
+    this._widgetDefinitionModel = opts.widgetDefinitionModel;
+  },
+
+  render: function () {
+    this.clearSubViews();
+    this.$el.html('');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-style-view.js]: Same here.

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js [hunk 4]
```diff
@@ -34,33 +34,41 @@ module.exports = cdb.core.View.extend({
...
-    // TODO: carousel -> Form Widget Type
     var self = this;
 
-    var types = WidgetFormFactory.getDataTypes(this._tableModel);
-
-    var $typeSelect = $(typesTemplate({ types: types }));
-    $typeSelect.change(function (newType) {
-      self._widgetDefinitionModel.changeType(this.value);
+    var tabPaneDataModel = new cdb.core.Model({
+      createContentView: function () {
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: Considering this kind of tab pane (text-labels) will be reused in various locations, why not use the semantic API `tabPaneFactory.createWithTextLabels` instead? That should remove a lot of the boilerplate here.

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-style-view.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    this._initViews();
+    return this;
+  },
+
+  _initViews: function () {
+    this.clearSubViews();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-style-view.js]: Double clearsubviews, not needed here.

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-data-view.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    this._initViews();
+    return this;
+  },
+
+  _initViews: function () {
+    this.clearSubViews();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-data-view.js]: Double clearSubViews, not needed here.

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js [hunk 7]
```diff
@@ -34,33 +34,41 @@ module.exports = cdb.core.View.extend({
...
     });
-    this.addView(formWidgetDataView);
-    this.$el.append(formWidgetDataView.render().el);
 
-    var formWidgetStyleView = new WidgetsStyleFormView({
-      widgetDefinitionModel: this._widgetDefinitionModel
+    this.$el.append(this.tabPaneView.render().$el);
+  },
+
+  _generateLabel: function (label) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: _genLabelView maybe better?

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js (hunk 4 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: Ey! yeah :dancer:

File: lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js (hunk 1 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/tab-pane/tab-pane-item-view.js]: True. I added a spec :)

File: lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js (hunk 4 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: :ok_hand:

Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: That should also remove the need for the private `_generateLabelView ` method etc.

Comment (javierarce) [inline lib/assets/javascripts/cartodb3/editor/widgets-form/widgets-form-content-view.js]: Changed

**Comment by javierarce** [comment]
Ok, suggestions implemented.

**Comment by xavijam** [comment]
Yeah! :+1:

**Pull request merged**